tools/scripts/list_stl_dependencies.sh

```diff
@@ -22,30 +22,74 @@
 #   ./list_stl_dependencies.sh
 #=====================================================
 
+SCOPE=${1:-list}
 COMPONENTS=(iceoryx_hoofs iceoryx_posh)
 SOURCE_DIR=(source include)
 WORKSPACE=$(git rev-parse --show-toplevel)
+QNX_PLATFORM_DIR=$WORKSPACE/iceoryx_hoofs/platform/qnx/
+USELIST=$WORKSPACE/tools/scripts/used-headers.txt
+TEMP_FILE=$(mktemp)
+GET_HEADER_NAME="\<\K[^<>]+(?=>)" # Matches the content between angle brackets
 
 for COMPONENT in ${COMPONENTS[@]}; do
     for DIR in ${SOURCE_DIR[@]}; do
         GREP_PATH="${GREP_PATH} ${WORKSPACE}/${COMPONENT}/$DIR"
     done
 done
 
+echo "# QNX platform/ libc headers" >> $TEMP_FILE
+
+echo "# usage of headers for QNX"
+gcc -w -fpreprocessed -dD -E $(find $QNX_PLATFORM_DIR -type f -iname *.cpp -o -iname *.hpp) \
+ | grep -e "#include <" \
+ | grep -oP $GET_HEADER_NAME \
+ | sort \
+ | uniq \
+ | tee -a $TEMP_FILE \
+ | cat
+
+
+echo "# iceoryx_posh / iceoryx_hoofs headers" >> $TEMP_FILE
+# GCC can't preprocess .inl so we grep them in plain text
 echo
-echo usage of std components by file
-grep -RIne "std::" $GREP_PATH | sed -n  "s/\([^:]*\:[0-9]*\)\:.*\(std::[a-zA-Z_]*\).*/\ \ \1  \2/p" | sort | uniq
+echo "# usage of <..> header includes"
+{ gcc -w -fpreprocessed -dD -E $(find $GREP_PATH -type f -iname *.cpp -o -iname *.hpp); cat $(find $GREP_PATH -type f -iname *.inl); } \
+ | grep -e "#include <" \
```

Comment:
This is not sufficient, here look at the wonderful diversity of c++.

```cpp
%:include <iostream> // this is legal, see digraphs
#include "iostream" // legal when you add the include path with `-I`
#include <iostream> // classic
%:include "iostream" // another possibility

#include          <iostream> // also legal, spaces for the win ... also in all the above allowed
#include <   iostream  > // this is illegal, no spaces in < > are allowed
#include <iostream> #include <vector> // this is legal c++ code but the second include is not processed
```

So this is the standard weirdness ... now lets get weirder and have more fun
```cpp
#define HYPNOTOAD <iostream>
#include HYPNOTOAD // hehehe this is legal

#define HASSEL_THE_HOFF <iostream> \
#include <vector>

#include HASSEL_THE_HOFF // woohooo it compiles and we have iostream and vector
```

Reply:
I am aware that we do not have to handle every weird corner case but extra spaces or " instead of < should be handled.

Reply:
Oh no why did you meantion ISO 646? :sweat_smile: You're completely correct with your analysis.! Instead of tackling every possible syntax in our little script, I suggest the following:

1. Run the alternative token check in `clang-tidy` and prevent any ISO 646 character from being used
2. `clang-format` ensures proper formatting
3. Preprocessor shenanigans like the ones you mentioned in the last paragraph should be found via a manual review